### ISSUE TITLE
Doors will eat objects

### ISSUE THREAD

**Issue by Noname12345GER** [MattJeanes/world-portals#14]
If you ram a car with opened doors, it goes into the interior with a 90% chance of crashing the server. Some sort of Filter for that?

**Comment by MattJeanes** [comment]
Transferring issue to world-portals as it is a bug in here

**Comment by LoganDark** [comment]
It might be possible to do something similar to what my portal system (for a different game engine) does. Teleport the entire creation at once only when the very center of it passes through the portal. Fast moving cars and etc. would be just fine. Not sure about sleds and such though, you might have to fake collisions if it's halfway through the portal, which is probably more effort than it's worth.

Teleporting one part at a time is a recipe for disaster unless you do some spoopy constraint overwriting that can somehow represent only half of a creation being on the other side of the portal. I don't know if that's possible in Gmod though...

**Comment by MattJeanes** [comment]
Yeah that's roughly what I was thinking, try and calculate the centre origin for the entire contraption, should work at least better than it does now

**Comment by AmyJeanes** [comment]
Still an issue - probably the biggest remaining issue for world-portals now is contraption teleportation.
